test/lint.js

```diff
@@ -0,0 +1,95 @@
+var fs = require('fs');
+var path = require('path');
+var Ajv = require('ajv');
+var ajv = new Ajv({ allErrors: true });
+var hasErrors = false;
+
+function jsonDiff(actual, expected) {
+  var actualLines = actual.split(/\n/);
+  var expectedLines = expected.split(/\n/);
+
+  for (var i = 0; i < actualLines.length; i++) {
+    if (actualLines[i] !== expectedLines[i]) {
+      return [
+        '#' + i + '\x1b[0m',
+        '    Actual:   ' + actualLines[i],
+        '    Expected: ' + expectedLines[i]
+      ].join('\n');
+    }
+  }
+}
+
+function checkStyle(filename) {
+  var actual = fs.readFileSync(filename, 'utf-8').trim();
+  var expected = JSON.stringify(JSON.parse(actual), null, 2);
+
+  if (actual === expected) {
+    console.log('\x1b[32m');
+    console.log('  Style – OK');
+    console.log('\x1b[0m');
+  } else {
+    hasErrors = true;
+    console.log('\x1b[31m');
+    console.log('  Style – Error on line ' + jsonDiff(actual, expected));
+  }
+}
+
+function checkSchema(dataFilename) {
+  var schemaFilename = '../compat-data.schema.json';
+  var valid = ajv.validate(
+    require(schemaFilename),
+    require(dataFilename)
+  );
+
+  if (valid) {
+    console.log('\x1b[32m');
+    console.log('  JSON schema – OK');
+    console.log('\x1b[0m');
+  } else {
+    hasErrors = true;
+    console.log('\x1b[31m');
+    console.log('  JSON schema – ' + ajv.errors.length + ' error(s)');
+    console.log('\x1b[0m');
+    console.log('   ' + ajv.errorsText(ajv.errors, {
+        separator: '\n    ',
+        dataVar: 'item'
+      })
+    );
+  }
+}
+
+
+function load() {
+  var dirAbs, dirRel = '';
+
+  function processFilename(fn) {
+    var fp = path.join(dirAbs, fn),
+    // If the given filename is a directory, recursively load it.
+    extra = fs.statSync(fp).isDirectory() ? load(fp) : require(fp);
+
+    if (path.extname(fn) === '.json') {
+      console.log(dirRel.replace('../', '') + '/' + fn);
+      checkStyle(fp)
+      checkSchema(fp);
+    }
+  }
+
+  for (dir of arguments) {
```

Comment:
Without an explicit ``let``, ``var``, or ``const`` in front of the ``dir``, what does JavaScript do here? It seems to work fine in any case. To be explicit, it might be nice to use ``const``?